internal/metrics/metric.go

```diff
@@ -147,7 +157,10 @@ func (m *Metric) GetDatum(labelvalues ...string) (d datum.Datum, err error) {
 			}
 			d = datum.NewBuckets(buckets)
 		}
-		m.LabelValues = append(m.LabelValues, &LabelValue{Labels: labelvalues, Value: d})
+		lv := &LabelValue{Labels: labelvalues, Value: d}
+		m.LabelValues = append(m.LabelValues, lv)
```

Comment:
Do we need both the map and the list now?

Reply:
Actuactually not need,  it's just for compatible. Drop the list maybe improve the performance further.